src/test/lib/shady-render_test.ts

```diff
@@ -22,50 +22,144 @@ const assert = chai.assert;
 
 suite('shady-render', () => {
 
-  test('prepares templates with ShadyCSS', () => {
+  test('styles elements rendered into shadowRoots', () => {
     const container = document.createElement('div');
+    document.body.appendChild(container);
+    container.attachShadow({mode: 'open'});
     render(html`
       <style>
         div {
-          color: red;
+          border: 2px solid blue;
         }
       </style>
       <div>Testing...</div>
-    `, container, 'scope-1');
-
-    assert.equal(container.children.length, 1);
-    const div = container.firstElementChild!;
-    assert.equal(div.getAttribute('class'), `style-scope scope-1`);
-    const style = document.querySelector('style[scope="scope-1"]');
-    assert.isNotNull(style);
+    `, container.shadowRoot as DocumentFragment, 'scope-1');
+    const div = (container.shadowRoot  as DocumentFragment).querySelector('div');
+    assert.equal(getComputedStyle(div as HTMLElement).getPropertyValue('border-top-width').trim(), '2px');
+    document.body.removeChild(container);
   });
 
-  test('prepares nested templates with ShadyCSS', () => {
+  test('styles elements rendred into shadowRoots in nested templates', () => {
     const container = document.createElement('div');
+    document.body.appendChild(container);
+    container.attachShadow({mode: 'open'});
     render(html`
       <style>
         div {
-          color: red;
+          border: 4px solid orange;
         }
       </style>
       <div>Testing...</div>
       ${html`
         <style>
           span {
-            color: blue;
+            border: 5px solid tomato;
           }
         </style>
         <span>Testing...</span>
       `}
-    `, container, 'scope-2');
+    `, container.shadowRoot as DocumentFragment, 'scope-2');
+    const div = (container.shadowRoot  as DocumentFragment).querySelector('div');
```

Comment:
the cast to DocumentFragment shouldn't be necessary

Reply:
done